book/crud.rst

```diff
@@ -0,0 +1,767 @@
+    define multiple ``dataBlocks`` here for this purpose.
+
+.. seealso::
+
+    Take a look at the templates that are shipped with the OroUIBundle to find out what blocks you
+    can also use to customize the layout of your forms.
+
+After the user submitted the form and the entered data is valid, the data will be persisted using
+the entity manager and a flash message is added which will be shown to the user on the request. The
+``oro_ui.router`` service is then used to redirect the user depending on the button they clicked to
+submit the form. They will be redirected to the first route that is passed to
+``redirectAfterSave()`` if they chose the *Save* button and if the user chose the *Save and close*
+button they will be redirected to the second route.
+
+.. _book-crud-view:
+
+Show Details of an Entity
+-------------------------
+
+To display all the details of an entity, you have to perform two steps. First, you need to create a
+controller that fetches the object from the database and passes it to a template:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Controller/VehicleController.php
+    namespace InventoryBundle\Controller;
+
+    use InventoryBundle\Entity\Vehicle;
+    use Oro\Bundle\SecurityBundle\Annotation\AclAncestor;
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+    // ...
+
+    class VehicleController extends Controller
+    {
+        /**
+         * @Route("/{id}", name="inventory.vehicle_view", requirements={"id"="\d+"})
+         * @Template
+         * @AclAncestor("inventory.vehicle_view")
+         */
+        public function viewAction(Vehicle $vehicle)
+        {
+            return array('vehicle' => $vehicle);
+        }
+    }
+
+And of course you also have to create the matching template:
+
+.. code-block:: html+jinja
+    :linenos:
+
+    {# src/InventoryBundle/Resources/views/Vehicle/view.html.twig #}
+    {% extends 'OroUIBundle:actions:view.html.twig' %}
+    {% import 'OroUIBundle::macros.html.twig' as UI %}
+
+    {% block navButtons %}
+        {% if resource_granted('EDIT', vehicle) %}
+            {{ UI.editButton({
+                'path' : path('inventory.vehicle_update', { id: vehicle.id }),
+                'entity_label': 'Vehicle'|trans
+            }) }}
+        {% endif %}
+
+        {% if resource_granted('DELETE', vehicle) %}
+            {{ UI.deleteButton({
+                'dataUrl': path('inventory_api_delete_vehicle', {'id': vehicle.id}),
+                'dataRedirect': path('inventory.vehicle_index'),
+                'aCss': 'no-hash remove-button',
+                'id': 'btn-remove-vehicle',
+                'dataId': vehicle.id,
+                'entity_label': 'Vehicle'|trans,
+            }) }}
+        {% endif %}
+    {% endblock navButtons %}
+
+    {% block pageHeader %}
+        {% set breadcrumbs = {
+            'entity':      form.vars.value,
```

Comment:
```form``` is not defined, ```entity``` should be used instead.
Also to render the breadcrumbs the Vehicle entity should have ```createdAt`` and ```updatedAt``` properties or corresponding getter methods. This should be fixed in the code example or at least simply noted in the documentation.